src/scripts/main.js

```diff
@@ -1,3 +1,276 @@
 'use strict';
 
-// write code here
+/* TABLE COLUMNS ASC DESC ORDER SORTING ____________________________________ */
+
+const heading = document.querySelector('thead').firstElementChild;
+const headingItems = [...heading.querySelectorAll('th')];
+const tableBody = document.querySelector('tbody');
+const tableRows = [...tableBody.children];
+
+headingItems.forEach(item => item.setAttribute('class', 'column'));
+
+for (const item of headingItems.slice(0, 3)) {
+  item.dataset.type = 'text';
+}
+headingItems[3].dataset.type = 'number';
+headingItems[4].dataset.type = 'salary';
+
+function tableSort(column) {
+  const index = headingItems.indexOf(column);
+
+  const callbackSort = (rowA, rowB) => {
+    const one = rowA.children[index].innerText;
+    const two = rowB.children[index].innerText;
+    let first = one;
+    let second = two;
+
+    if (column.dataset.order === 'asc') {
+      first = two;
+      second = one;
+    }
+
+    const transformSalary = (cell) => {
+      return cell.slice(1).split(',').join('');
+    };
+
+    if (column.dataset.type === 'text') {
+      return first.localeCompare(second);
+    } else if (column.dataset.type === 'salary') {
+      return transformSalary(first)
+      - transformSalary(second);
+    } else if (column.dataset.type === 'number') {
+      return first - second;
+    }
+  };
+  const sorted = [...tableBody.children].sort(callbackSort);
+
+  tableRows.forEach(item => item.remove());
+  sorted.forEach(item => tableBody.append(item));
+
+  if (!column.hasAttribute('data-order')
+  || column.dataset.order === 'desc') {
```

Comment:
pass big conditions (`!column.hasAttribute('data-order')
  || column.dataset.order === 'desc'`) into separate variable and use here